dispatch/dispatch.go

```diff
@@ -128,27 +128,31 @@ func (d *Dispatcher) stop() error {
 	// Release finished workers
 	close(d.shutdown)
 
+	ch := make(chan struct{})
```

Comment:
can buffer this as the routine can leak below if there is no reader

Reply:
Good point.
Fixed 6defb7dc